client.go

```diff
@@ -581,10 +589,10 @@ func (cli *Client) executeCompiledRequest(req *http.Request, retries int, backof
 	var body []byte
 	if res.StatusCode < 200 || res.StatusCode >= 300 {
 		body, err = ParseErrorResponse(req, res)
-		cli.LogRequestDone(req, res, nil, len(body), duration)
+		cli.LogRequestDone(req, res, err, nil, len(body), duration)
```

Comment:
Bad HTTP status codes shouldn't be logged as errors at this layer since they might not be errors

```suggestion
		cli.LogRequestDone(req, res, nil, nil, len(body), duration)
```